pkg/controller/controller_instance.go

```diff
@@ -1659,6 +1803,12 @@ func (c *controller) handleServiceInstanceReconciliationError(instance *v1beta1.
 // processServiceInstanceOperationError handles the logging and updating of
 // a ServiceInstance that hit a retryable error during reconciliation.
 func (c *controller) processServiceInstanceOperationError(instance *v1beta1.ServiceInstance, readyCond *v1beta1.ServiceInstanceCondition) error {
+	// If error is from an update, assume a retry will happen, set a not-before time so we
+	// don't pound the Broker in a constant update/fail/update repeat loop.
+	if strings.Contains(readyCond.Reason, "UpdateInstance") {
```

Comment:
so instead of setting retry time under condition, we might just set the flag "delay required" there (unconditionally), and set the delay in the `reconcileServiceInstanceUpdate` if the flag is set.